lib/pc_patch_uncompressed.c

```diff
@@ -222,6 +222,15 @@ pc_patch_uncompressed_free(PCPATCH_UNCOMPRESSED *patch)
 	pcfree(patch);
 }
 
+// Makes the patch readonly. Returns the memory segment
+// owned by the patch, if any, to enable transfer of ownership
+uint8_t *
+pc_patch_uncompressed_readonly(PCPATCH_UNCOMPRESSED *patch)
+{
+	uint8_t *data = patch->readonly ? NULL : patch->data;
+	patch->readonly = PC_TRUE;
+	return data;
+}
```

Comment:
This was just a suggestion. It's a matter of taste. Keep your version if you like it best.